akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -431,16 +435,19 @@ private[stream] final class GraphInterpreterShell(
 
   private val resume = Resume(this)
 
-  private def runBatch(): Unit = {
+  private var currentLimit = effectiveLimit()
```

Comment:
shouldn’t this be decremented somewhere? as currently coded this does not make sense to me